private/protocol/json/jsonutil/unmarshal.go

```diff
@@ -187,9 +187,12 @@ func (u unmarshaler) unmarshalList(value reflect.Value, data interface{}, tag re
 		return fmt.Errorf("JSON value is not a list (%#v)", data)
 	}
 
-	if value.IsNil() {
-		l := len(listData)
+	if l := len(listData); value.IsNil() {
 		value.Set(reflect.MakeSlice(value.Type(), l, l))
+	} else {
+		if value.Kind() != reflect.Slice || value.Len() < len(listData) {
```

Comment:
Perhaps we could simply replace the list with a correctly sized one, rather than emitting an error.​